server/modules/survey/manager/surveyManager.js

```diff
@@ -185,16 +190,13 @@ const updateSurveyProps = async (user, surveyId, props, client = db) =>
     }
   })
 
-const publishSurveyProps = async (surveyId, langsDeleted, client = db) => {
-  await SurveyRepository.publishSurveyProps(surveyId, client)
-
-  for (const langDeleted of langsDeleted) {
-    await Promise.all([
-      SurveyRepository.deleteSurveyLabel(surveyId, langDeleted, client),
-      SurveyRepository.deleteSurveyDescription(surveyId, langDeleted, client),
-    ])
-  }
-}
+const publishSurveyProps = async (surveyId, langsDeleted, client = db) =>
+  client.tx(async t => {
+    const surveyUpdated = SurveyRepository.publishSurveyProps(surveyId, t)
+    if (!R.isEmpty(langsDeleted))
+      await SurveyRepository.deleteSurveyLabelsAndDescriptions(surveyId, langsDeleted, t)
+    return surveyUpdated
```

Comment:
this method before didn't return anything. is the return needed?

Reply:
the return is needed from outside to avoid fetching again the survey to get the survey uuid